contracts/staking/StakeManager.sol

```diff
@@ -190,18 +219,20 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     _unstake(validatorId, exitEpoch);
   }
 
-  function delegationTransfer(uint256 amount, address delegator) external {
-    require(Registry(registry).getDelegationManagerAddress() == msg.sender);
+  function delegationTransfer(uint256 validatorId, uint256 amount, address delegator) external {
+    require(validators[validatorId].contractAddress == msg.sender);
     require(token.transfer(delegator, amount), "Insufficent rewards");
   }
 
-  function stakeFor(address user, uint256 amount, address signer, bool isContract) public onlyWhenUnlocked {
+  function stakeFor(address user, uint256 amount, uint256 heimdallFee, address signer, bool isContract) public onlyWhenUnlocked {
     require(currentValidatorSetSize() < validatorThreshold);
     require(balanceOf(user) == 0, "Only one time staking is allowed");
     require(amount > MIN_DEPOSIT_SIZE);
```

Comment:
Minimum deposit size is just 1 token even for staking?

Reply:
yes because we have replacement mech.